### PR TITLE
APIGOV-19801 - change some debug logs to trace

### PR DESCRIPTION
**Pull request by dfeldick**
Based on the ambiguity of debug vs trace and what customers want or need to see, this is about the extent I want to go for right now. Just move some of the messages that are clearly pretty low-level. If there are no objections and this is approved, I will take a similar approach to the agents. If you feel that it's gone overboard, let me know also.

### CODE REVIEW

File: docs/utilities/index.md [hunk 1]
```diff
@@ -186,6 +186,8 @@ log.Infof("Log entry with format, %s", "additional log ~~message")
 
 log.Debugf("No changes detected in the API %s", *azAPI.Name)
 
+log.Trace("I got here in the code")
+
```
Comment (sbolosan) [inline docs/utilities/index.md:190]: what is this for?

File: pkg/cmd/root.go [hunk 2]
```diff
@@ -191,7 +191,7 @@ func (c *agentRootCommand) initialize(cmd *cobra.Command, args []string) error {
 
 	viper.WatchConfig()
 	viper.OnConfigChange(func(e fsnotify.Event) {
-		log.Debugf("Config file changed : %s", e.Name)
+		log.Tracef("Config file changed : %s", e.Name)
```
Comment (sagarkal) [inline pkg/cmd/root.go:194]: I think this needs to be info, because its config change

File: docs/utilities/index.md (hunk 1 above)
Comment (dfeldick) [reply docs/utilities/index.md:190]: this is a file showing examples of how to use the log calls. There was no previous example of how to use Trace if a customer wanted to do such a thing in their agents.

Comment (sbolosan) [reply docs/utilities/index.md:190]: ok...

**Comment by sbolosan** [comment]
Good start...  i know i see some i created, so i'm guilty of abusing debug...

File: pkg/cmd/root.go (hunk 2 above)
Comment (dfeldick) [reply pkg/cmd/root.go:194]: Cyrille, in another ticket, had a specific list of what should be info. This wasn't one. But maybe it should stay debug? I don't know.

**Review by sagarkal — APPROVED** [review]

File: pkg/cmd/root.go (hunk 2 above)
Comment (sbolosan) [reply pkg/cmd/root.go:194]: I think Info or debug would work... since @sagarkal  is right, I think stuff like config changed should be seen by customer

**Review by jcollins-axway — APPROVED** [review]

File: pkg/cmd/root.go (hunk 2 above)
Comment (dfeldick) [reply pkg/cmd/root.go:194]: put back to debug

**Pull request merged**